### PR TITLE
ankama-launcher: init at 3.12.24

### PR DESCRIPTION
**Pull request by Harbiinger**
## Description of changes
Added a new package: ankama-launcher. Used to play Ankama games like the mmorpg Dofus.
( new version of  #336028 )

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Harbiinger** [comment]
Sorry I had to remake a PR. I closed the last one by accident and had problems with git.
I applied @SigmaSquadron 's and FilipeS0usa's advices

**Comment by FilipeS0usa** [comment]
> Sorry I had to remake a PR. I closed the last one by accident and had problems with git. I applied @SigmaSquadron 's and @FilipeS0usa 's advices

If you don't want to use the URL that I told you was available, you could just use what you had, it also works. I build it and it worked well. 

And I don't know if it's relevant or not, but, the only thing that didn't work was running Dofus and Dofus Beta. In Dofus Beta I could open the game but then I couldn't enter any server. I don't know if you have the same problem or not, or if it's even possible to fix it in the package.  (This was for the two versions of the package, the one that you made and the one that I made)

**Comment by Harbiinger** [comment]
> the only thing that didn't work was running Dofus. 

Dofus is working for me. Did you rename or delete the cinematics directory ? I wrote the instructions in the longDescription.

> In Dofus Beta I could open the game but then I couldn't enter any server. I don't know if you have the same problem or not.

Yes I have the same problem I can't enter any server even though I have an active subscription. I tried both port 443 and 5555.

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  lib,
+}:
+let
+  pname = "ankama-launcher";
+  version = "3.12.17";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://launcher.cdn.ankama.com/installers/production/Ankama%20Launcher-Setup-x86_64.AppImage";
+    hash = "sha256-aD5YYcye/WPpb0Kir55qCL+RGbK8ZSdmi4NKmqKkW/Y=";
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:13]: Hash needs to be updated.
```suggestion
        hash = "sha256-JERXkIp0psOu5L3bSpuOgfxokZ1pB4/BHpL2np2aHWo=";
```

**Comment by FilipeS0usa** [comment]
I've been having a lot of work, I'm going to try to put it running this weekend. I still had some problems with Dofus and Dofus Beta.

- Dofus Beta still had the problem with the servers. 
- And with Dofus I still couldn't enter the game, I'm going to rebuild and remove the cinematics directory again to see if it can run.

**Comment by SigmaSquadron** [comment]
While this is somewhat frowned upon, as this is a proprietary project, we may need to use an FHS here.

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  appimageContents = appimageTools.extract { inherit name src; };
+in
+
+appimageTools.wrapType2 rec {
+  inherit name src;
+  extraPkgs = pkgs: [ pkgs.wine ];
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:21]: I don't know if we really need this line. Are we using wine?
```suggestion
```

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  extraInstallCommands = ''
+    install -m 444 -D ${appimageContents}/zaap.desktop $out/share/applications/ankama-launcher.desktop
+    sed -i 's/.*Exec.*/Exec=ankama-launcher/' $out/share/applications/ankama-launcher.desktop
+    install -m 444 -D ${appimageContents}/zaap.png $out/share/icons/hicolor/256x256/apps/zaap.png
+  '';
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:27]: What is the real reason for us to do this extraInstallCommands?

**Review by FilipeS0usa — COMMENTED** [review]
I'm trying to put this working again. Let's try to see everything that we are doing to discover what we are doing wrong.

**Comment by FilipeS0usa** [comment]
> While this is somewhat frowned upon, as this is a proprietary project, we may need to use an FHS here.

I think we are already using FHS. When we call `appimageTools.wrapType2 {};` we are creating a FHS environment you can check the [documentation](https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-appimageTools).

**Comment by FilipeS0usa** [comment]
Good news I think the problem is fixed!!!!!

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  fetchurl,
+  lib,
+}:
+let
+  pname = "ankama-launcher";
+  version = "3.12.17";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://launcher.cdn.ankama.com/installers/production/Ankama%20Launcher-Setup-x86_64.AppImage";
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:12]: Change the url for the following one:
```suggestion
    url = "https://launcher.cdn.ankama.com/installers/production/Dofus-Setup-x86_64.AppImage";
```
I found this one inspecting the Dofus page on the network tab. I clicked the download button while inspecting the network and got this request URL eheheheh. And it works perfectly :)

File: pkgs/by-name/an/ankama-launcher/package.nix (hunk 1 above)
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:13]: The corresponding hash should be the following:
```suggestion
    hash = "sha256-ssCjFrQYI/pxnsjcy1xLSpuGGy0NGMOcNP9RiNDhE/w=";
```

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    hash = "sha256-aD5YYcye/WPpb0Kir55qCL+RGbK8ZSdmi4NKmqKkW/Y=";
+  };
+
+  appimageContents = appimageTools.extract { inherit name src; };
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:16]: You don't need this line
```suggestion
```

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+in
+
+appimageTools.wrapType2 rec {
```
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:19]: You don't need `rec`
```suggestion
appimageTools.wrapType2 {
```

File: pkgs/by-name/an/ankama-launcher/package.nix (hunk 3 above)
Comment (FilipeS0usa) [inline pkgs/by-name/an/ankama-launcher/package.nix:27]: You don't need this extraInstallCommands
```suggestion
```

**Review by FilipeS0usa — COMMENTED** [review]
All of these changes should put the package working perfectly. Try to do the changes and build the package. Tell me if it works for you.

With this changes you don't need to change any file/directory. You just build it and play :100: 

image

**Comment by Harbiinger** [comment]
> All of these changes should put the package working perfectly. Try to do the changes and build the package. Tell me if it works for you.
It works perfectly for me 
> With this changes you don't need to change any file/directory. You just build it and play 💯
You still need to rename or remove the cinematics directory if you want to create a new character in Dofus.

File: pkgs/by-name/an/ankama-launcher/package.nix (hunk 3 above)
Comment (Harbiinger) [reply pkgs/by-name/an/ankama-launcher/package.nix:26]: It is not necessary but I prefer to have a desktop file to launch it

**Comment by Harbiinger** [comment]
> All of these changes should put the package working perfectly. Try to do the changes and build the package. Tell me if it works for you.
> 
> With this changes you don't need to change any file/directory. You just build it and play 💯
> 
> image

Do you want to add yourself to the maintainers list ?

**Comment by FilipeS0usa** [comment]
> You still need to rename or remove the cinematics directory if you want to create a new character in Dofus.

True, just got that error, changed the name and worked. Do you think there is a way to automate the process of removing/renaming the directory? (I think it's not possible because you need to first install Dofus do be able to have access to that directory) But also when they make the port to Unity on Dofus, it will probably resolve this issue.

> Do you want to add yourself to the maintainers list ?

Sure, but I think I don't yet have the knowledge to be a maintainer of a nix package, and I still don't know what that implies. I still have a lot to learn, eheheheh.

@

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  lib,
+}:
+let
+  pname = "ankama-launcher";
+  version = "3.12.17";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://launcher.cdn.ankama.com/installers/production/Dofus-Setup-x86_64.AppImage";
+    hash = "sha256-ssCjFrQYI/pxnsjcy1xLSpuGGy0NGMOcNP9RiNDhE/w=";
```
Comment (GaetanLepage) [inline pkgs/by-name/an/ankama-launcher/package.nix:13]:
```suggestion
    hash = "sha256-SfK8zfWcxpILi6VGK72ja0WgIDIvZ4nOR70ToEt9rMk=";
```

**Review by NickCao — APPROVED** [review]
Diff LGTM.

**Comment by JohnRTitor** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 337438`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ankama-launcher</li>
  </ul>
</details>

**Comment by noahfraiture** [comment]
Hello, on start I get this error 
image

With these logs : 
```


12/26/2024, 12:50:50 PM.415 - info: [MAIN] setup
12/26/2024, 12:50:50 PM.421 - info: [TWITCH] setup
12/26/2024, 12:50:50 PM.424 - info: [APP] setup done
12/26/2024, 12:50:50 PM.431 - info: [MAIN] App ready - version 3.12.27
12/26/2024, 12:50:50 PM.431 - info: [MAIN] Main window opened
12/26/2024, 12:50:50 PM.441 - info: [CONNECTIVITY] setup
12/26/2024, 12:50:50 PM.638 - info: [CONNECTIVITY] navigator-online-changed true
12/26/2024, 12:50:50 PM.659 - info: [CONNECTIVITY] We are now online
12/26/2024, 12:50:50 PM.768 - info: [VUE MAIN] Mounting
12/26/2024, 12:50:50 PM.792 - info: [MAIN] Zaap main window is Ready
12/26/2024, 12:50:50 PM.793 - info: [AUTO_UPDATER] autoUpdate setup
12/26/2024, 12:50:50 PM.801 - info: [AUTO_UPDATER] feed url set to: https://launcher.cdn.ankama.com/installers/production
12/26/2024, 12:50:50 PM.880 - info: [AUTO_UPDATER] Has checked for updates true
12/26/2024, 12:50:50 PM.880 - info: [AUTO_UPDATER] An update is available
12/26/2024, 12:50:50 PM.884 - error: [1015 AUTO_UPDATER] error, Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:184:186147)
    at d.executeDownload (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:71:80166) Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:184:186147)
    at d.executeDownload (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:71:80166)
12/26/2024, 12:50:50 PM.884 - error: [MAIN] Autoupdater failed
12/26/2024, 12:50:50 PM.884 - info: [MAIN] primarySetups
12/26/2024, 12:50:50 PM.885 - error: [1074 MAIN] A fatal error occurred (unhandled rejection) Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:184:186147)
    at d.executeDownload (/nix/store/fvgrx8sabnkglz6hlr71sgv9fcs8vm36-ankama-launcher-3.12.27-extracted/resources/app.asar/main.js:71:80166)
12/26/2024, 12:50:50 PM.894 - error: [1083 SENTRY] Sending error data to sentry, event id: 60402d314aa444e9b70572fd34a4de54
12/26/2024, 12:50:50 PM.901 - info: [I18next] Successfully initialized
12/26/2024, 12:50:50 PM.901 - info: [UNIVERSE] setup
12/26/2024, 12:50:50 PM.978 - info: [UNIVERSE] Downloaded universes config
12/26/2024, 12:50:50 PM.990 - info: No settings for service webtoon in universe /home/noah/.config/zaap/universe/Allskreen
12/26/2024, 12:50:50 PM.990 - info: No settings for service shop in universe /home/noah/.config/zaap/universe/Allskreen
12/26/2024, 12:50:50 PM.996 - info: No settings for service stream in universe /home/noah/.config/zaap/universe/Krosmoz
12/26/2024, 12:50:50 PM.997 - info: No settings for service shop in universe /home/noah/.config/zaap/universe/Krosmoz
12/26/2024, 12:50:50 PM.997 - info: No settings for service webtoon in universe /home/noah/.config/zaap/universe/Krosmoz
12/26/2024, 12:50:50 PM.997 - info: No settings for service video in universe /home/noah/.config/zaap/universe/Krosmoz
12/26/2024, 12:50:51 PM.10 - info: [REGISTRY] setup
12/26/2024, 12:50:51 PM.21 - warn: [ZB-514] setConfiguration (load)
12/26/2024, 12:50:51 PM.23 - info: [RELEASE]  dofus release must be repaired
12/26/2024, 12:50:51 PM.50 - info: [UNIVERSE] Register universe Krosmoz
12/26/2024, 12:50:51 PM.50 - info: [UNIVERSE] Register universe Allskreen
12/26/2024, 12:50:51 PM.51 - info: [UNIVERSE] krosmoz selected
12/26/2024, 12:50:51 PM.85 - info: [AUTH] Sign on user
12/26/2024, 12:50:51 PM.522 - info: [TEXTSOCKETSERVER] starting server on port 26117
12/26/2024, 12:50:51 PM.524 - info: [SHOP] setup
12/26/2024, 12:50:51 PM.526 - info: [PIP] setup
12/26/2024, 12:50:51 PM.526 - info: [OVERLAY] setup
12/26/2024, 12:50:51 PM.527 - info: [AUTO_LAUNCH] setting to false
12/26/2024, 12:50:51 PM.527 - info: [NEWS] setup
12/26/2024, 12:50:51 PM.527 - info: [CHANGELOGS] setup
12/26/2024, 12:50:51 PM.527 - info: [TERMS] setup
12/26/2024, 12:50:51 PM.529 - info: [SERVICE] started server on port 26116
12/26/2024, 12:50:51 PM.529 - info: [TEXTSOCKETSERVER] started server on port 26117
12/26/2024, 12:50:51 PM.835 - info: [CHAT] Chat connected
12/26/2024, 12:50:51 PM.837 - info: [CHAT] Chat initialized
```

is it normal ?

**Comment by c-leri** [comment]
I encountered the same issue, for some reason you need to set the APPIMAGE env variable when launching the launcher for the first time, it works fine after that

**Comment by noahfraiture** [comment]
> I encountered the same issue, for some reason you need to set the APPIMAGE env variable when launching the launcher for the first time, it works fine after that

Indeed it solves the problem. Thank you

**Comment by Harbiinger** [comment]
Maybe there's a way to set the variable in the package.nix.
I'll look into that tomorrow if I have the time

**Comment by luvcie** [comment]
```
export APPIMAGE=/dev/null
```
This tricks the launcher into thinking it's running as an AppImage.
Then you just launch it, and it should work.

**Comment by luvcie** [comment]
I also got that error with a log like that.

```
export APPIMAGE=/dev/null
```
As a temporary workaround, this tricks the launcher into thinking it's running as an AppImage.
Then you just launch it and it should work fine.

**Comment by luvcie** [comment]
> I encountered the same issue, for some reason you need to set the APPIMAGE env variable when launching the launcher for the first time, it works fine after that

I'm having the same issue. What do you set the env variable as?
As a test, when I put any location at the end of this command, it lets me load the launcher, but then it doesn't let me download Dofus.
```
export APPIMAGE=
```

**Comment by HouseDwarfSpace** [comment]
I am also experiencing the same issue. After `export APPIMAGE=1`, the launcher manages to start and I can login and choose which game to download. But the download never actually happens - if I click on the download Details I get an indefinite _Connecting with the Krosmoz_ message. I'm not using any fancy DNS settings - this is the only snippet of nix code that takes care of networking for me:

```
networking = {
    hostName = "nixos";
    networkmanager.enable = true;
}
```

From the logs that `ankama-launcher` spits to stdout, I see:

```
23/03/2025, 18:01:29.875 - info: [KPI] gameUpdate 
{
  type: 'UPGRADE',
  funnelStep: 'HIT',
  gameUid: 'dofus',
  releaseName: 'dofus3',
  autoUpgrade: true,
  sourceVersion: false,
  targetVersion: undefined,
  reason: 'repair_hit',
  downloadSpeed: undefined,
  updateSize: undefined,
  updateFiles: undefined
}
23/03/2025, 18:01:29.876 - debug: [KPI] event 677 added
23/03/2025, 18:01:29.876 - info: [RELEASE_UPDATE] createUpdate dofus dofus3
23/03/2025, 18:01:30.322 - warn: [ZB-514] setConfiguration (load)
23/03/2025, 18:01:30.332 - debug: [RELEASE HOOK] hook not defined in configuration, skipping 
{ release: 'dofus3', gameUid: 'dofus', hook: 'settings_changed' }
23/03/2025, 18:01:30.340 - debug: [GAME_CONFIG] New configuration dofus dofus3
23/03/2025, 18:01:30.342 - warn: Cannot start update, queue is paused
23/03/2025, 18:01:30.342 - debug: [RELEASE HOOK] hook not defined in configuration, skipping 
{ release: 'dofus3', gameUid: 'dofus', hook: 'pre_update' }
23/03/2025, 18:01:30.342 - info: [KPI] gameUpdate 
{
  type: 'REPAIR',
  funnelStep: 'START',
  gameUid: 'dofus',
  releaseName: 'dofus3',
  autoUpgrade: true,
  sourceVersion: false,
  targetVersion: '6.0_3.0.50.47',
  reason: 'dirty release',
  downloadSpeed: NaN,
  updateSize: 0,
  updateFiles: 0
}
23/03/2025, 18:01:30.343 - debug: [KPI] event 677 added
23/03/2025, 18:01:30.501 - debug: [KPI] 5 event(s) sent
```

**Comment by HouseDwarfSpace** [comment]
I'm willing to bet it's my nixos setup that is the issue, probably unrelated to this package. In the meantime I tried https://github.com/Antiz96/Ankama-Launcher-Container/tree/main and managed to get the download to start through the ankama-launcher bin offered by the container.

**Comment by Aire-One** [comment]
Hello, thank you for the work here! I have installed this package from the unstable branch (the launcher displays `v3.12.38`). I have similar issues than what was reported by others:

1. The launcher fails to start until I open it from the terminal with `APPIMAGE= ankama-launcher`.
2. Games cannot be installed, and keep displaying a "PRIORITIZE" button.

In the logs, there is the following error:

```
4/13/2025, 5:44:42 PM.489 - error: [1015 AUTO_UPDATER] error, Error: EROFS: read-only file system, unlink '/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/AppRun'
    at Object.unlinkSync (node:fs:1718:3)
    at doInstall (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:187044)
    at d.install (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:188547)
    at d.quitAndInstall (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:187827)
    at s.restartForUpdate (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:46595)
    at s.updateLauncher (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:46331)
    at d.<anonymous> (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:45394)
    at d.emit (node:events:394:28)
    at d.emit (node:domain:475:12)
    at dispatchUpdateDownloaded (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:71:77512) Error: EROFS: read-only file system, unlink '/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/AppRun'
    at Object.unlinkSync (node:fs:1718:3)
    at doInstall (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:187044)
    at d.install (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:188547)
    at d.quitAndInstall (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:184:187827)
    at s.restartForUpdate (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:46595)
    at s.updateLauncher (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:46331)
    at d.<anonymous> (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:13:45394)
    at d.emit (node:events:394:28)
    at d.emit (node:domain:475:12)
    at dispatchUpdateDownloaded (/nix/store/ymfn05cvbcd4wp425ah14ayaw3d9cv5d-ankama-launcher-3.12.38-extracted/resources/app.asar/main.js:71:77512)
```

So I guess the launcher tries to self update, but Nix Store immutability prevent it to proceed.

When I click on the "PRIORITIZE" button, I get the log:

```
4/13/2025, 5:46:46 PM.503 - warn: Cannot start update, queue is paused
```

By going through the menus to start a "repair" of the game, I get this more explicit log:

```
4/13/2025, 5:58:18 PM.0 - error: [1053 RELEASE] Error: Cannot update, an update is already running
```

I couldn't find a way to prevent the launcher from trying to self update.

Does the source URL from web.archive.org need to be updated for the launcher to be up-to-date and allow game's download?

**Pull request merged**